### PR TITLE
Use keyvault_id instead of vault uri 

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

- Currently on prod build fails with below error
```* data.azurerm_key_vault_secret.cert: data.azurerm_key_vault_secret.cert: Unable to locate the Resource ID for the Key Vault at URL "https://infra-cert-prod.vault.azure.net/": %!s(<nil>)```

- This is an attempt to fix the issue as `vault_uri` is already deprecated and hence using `key_vault_id`

- Working build on sandbox https://sandbox-build.platform.hmcts.net/job/HMCTS_Sandbox_Platform/job/bulk-scan-shared-infrastructure/job/USE-KEYVAULT-ID/

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: app-gateway.tf [hunk 1]
```diff
@@ -1,6 +1,21 @@
+locals {
+  prod_vault      = "${(var.env == "prod") ? "infra-vault-prod" : ""}"
+  nonprod_vault   = "${(var.env == "demo" || var.env == "aat") ? "infra-vault-nonprod" : ""}"
+  sandbox_vault   = "${(var.env == "sandbox" || var.env == "saat" || var.env == "sprod") ? "infra-vault-sandbox" : ""}"
+
+  vaultNameIfSubscriptionPresent = "infra-vault-${var.subscription}"
+
+  vaultName = "${var.subscription != "" ? local.vaultNameIfSubscriptionPresent : format("%s%s%s%s", local.prod_vault, local.nonprod_vault, local.sandbox_vault)}"
```
Comment (timja) [inline app-gateway.tf]: you don't need this the subscription will always be present for you

Comment (nitinprabhu) [reply app-gateway.tf]: done thanks

File: app-gateway.tf [hunk 2]
```diff
@@ -1,6 +1,15 @@
+locals {
+  vaultName = "${(var.env == "demo" || var.env == "aat") ? "infra-vault-nonprod" : (var.env == "sandbox" || var.env == "saat" || var.env == "sprod") ? "infra-vault-sandbox" : var.env == "prod" ? "infra-vault-prod" : ""}"
```
Comment (timja) [inline app-gateway.tf]: you don't need this

File: app-gateway.tf [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+}
+
+data "azurerm_key_vault" "infra_vault" {
+  name = "${local.vaultName}"
```
Comment (timja) [inline app-gateway.tf]:
```suggestion
  name = "infra-vault-${var.subscription}"
```

**Pull request merged**